src/Reflection/ReflectionClass.php

```diff
@@ -584,9 +586,11 @@ static function (ReflectionClassConstant $constant): string {
      * Get an associative array of the defined constants in this class,
      * with keys as constant names and values as {@see ReflectionClassConstant} objects.
      *
+     * @see ReflectionClass::getConstants() for the usage of filter
+     *
      * @return array<string, ReflectionClassConstant> indexed by name
      */
-    public function getReflectionConstants(): array
+    public function getReflectionConstants(?int $filter = null): array
```

Comment:
I don't think the `$filter` as a bitmask is a good API: sure, it perhaps worked in php-src, but it's a very much shitty approach that C programmers keep perpetuating and imposing upon sensible developers and scopes where memory utilization isn't needed to this level.

I think this `array_filter()` should be done in `Adapter\ReflectionClass#getReflectionConstants()` instead.